### PR TITLE
Fully implemented client/init and client/task endpoints

### PR DESCRIPTION
**Pull request by Rickebo**
Both `client/init` and `client/task` endpoints are now fully implemented. This PR contains changes mainly related to:
- Storing client data in database, under users collection.
- Storing client tasks in database, under user_tasks collection.
- Storing done client tasks in database, under user_done_tasks collection.

Tests are also added for the database, and the old tests were changed reflect the changes done to both endpoints.

### CODE REVIEW

File: client/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+cmake_minimum_required(VERSION 3.20)
```
Comment (johanlovgren) [inline client/CMakeLists.txt:1]: This file should be deleted! 
Make sure you fetch changes from master that removes all C++ related files before merging to master

File: client/main.cpp [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#include <iostream>
```
Comment (johanlovgren) [inline client/main.cpp:1]: This file should be deleted! 
Make sure you fetch changes from master that removes all C++ related files before merging to master

**Review by johanlovgren — APPROVED** [review]
Nice job! Pushing out code like a machine.

File: client/CMakeLists.txt (hunk 1 above)
Comment (Rickebo) [reply client/CMakeLists.txt]: Fixed in [aa99708](https://github.com/ghmo2789/puppet-master/pull/59/commits/aa99708d2fbf1aacfaa397298f3f5aec65dab5f3)

File: client/main.cpp (hunk 2 above)
Comment (Rickebo) [reply client/main.cpp]: Fixed in [aa99708](https://github.com/ghmo2789/puppet-master/pull/59/commits/aa99708d2fbf1aacfaa397298f3f5aec65dab5f3)

**Review by johanlovgren — APPROVED** [review]

**Pull request merged**